### PR TITLE
RHCLOUD-30986 | feature: principals with User Access Admin role should manage groups

### PR DESCRIPTION
**Pull request by MikelAlejoBR**
## Link(s) to Jira
- [[RHCLOUD-30986]](https://issues.redhat.com/browse/RHCLOUd-30986)

## Description of Intent of Change(s)
The service accounts and user based principals with the 'User Access administrator' role should be able to create, modify and delete groups.

### Things checked

Users within a 'User Access administrator' role can:

- Groups. 
  - Create a group.
  - Modify a group.
  - Delete a group.
- Roles.
  - Add a role to a group.
  - Delete a role from a group.
- Principals.
  - Add a principal to a group.
  - Remove a principal from a group.

Users within a 'User Access administrator' role can **but should not be able to:**

- Groups.
  - Add and remove principals from groups with the User Administrator Role. 
  - Update a group with an associated User Administrator Role.
  - Delete a group with an associated User Administrator Role.

### How can the feature be exercised?
1. Create a group with the User Access Administrator role.
2. Create an `x-rh-identity` header for a service account and another one for a user based principal.
3. Use them to attempt to add or remove principals to a group, to create a group, delete it, etc.

Right now, if this is tested in stage, the principals are not allowed to manage groups even if they have the right role assigned to it.

-----------
Because this covers what principals can or cannot do with /groups/* endpoints, we prepared audit for this endpoint and acceptance criteria checklist - details in [RHCLOUD-31456](https://issues.redhat.com/browse/RHCLOUD-31456)

## Checklist
- [ ] if API spec changes are required, is the spec updated?
- [ ] are there any pre/post merge actions required? if so, document here.
- [x] are theses changes covered by unit tests?
- [ ] if warranted, are documentation changes accounted for?
- [ ] does this require migration changes?
  - [ ] if yes, are they backwards compatible?
- [ ] is there known, direct impact to dependent teams/components?
  - [ ] if yes, how will this be handled?

## Secure Coding Practices Checklist Link
- https://github.com/RedHatInsights/secure-coding-checklist

## Secure Coding Practices Checklist
- [ ] Input Validation
- [ ] Output Encoding
- [ ] Authentication and Password Management
- [ ] Session Management
- [ ] Access Control
- [ ] Cryptographic Practices
- [ ] Error Handling and Logging
- [ ] Data Protection
- [ ] Communication Security
- [ ] System Configuration
- [ ] Database Security
- [ ] File Management
- [ ] Memory Management
- [ ] General Coding Practices

### CODE REVIEW

**Comment by petracihalova** [comment]
rebased & conflicts solved

**Review by MikelAlejoBR — COMMENTED** [review]
LGTM! Really good job!

File: tests/management/group/test_view.py [hunk 1]
```diff
@@ -2679,3 +2789,1797 @@ def test_group_role_filter_as_non_admin(self):
...
+        test_group = Group(name="test group", tenant=self.tenant)
+        test_group.save()
+
+        url = reverse("group-detail", kwargs={"uuid": test_group.uuid})
+        response = client.delete(url, **self.headers_service_account_principal)
+        self.assertEqual(response.status_code, status.HTTP_204_NO_CONTENT)
+
+    def test_remove_group_with_User_Access_Admin_fail(self):
+        """
+        Test that non org admin with 'User Access administrator' role cannot update a group
```
[the remark this answers is not in the document]
Comment (petracihalova) [reply tests/management/group/test_view.py:3532]: nice catch !!!

[the remark this answers is not in the document]
Comment (petracihalova) [reply tests/management/group/test_view.py]: fixed

File: tests/management/group/test_view.py [hunk 2]
```diff
@@ -2679,3 +2789,1798 @@ def test_group_role_filter_as_non_admin(self):
...
+        service account based principal into a group with 'User Access administrator' role.
+        """
+        # Create a group with 'User Access administrator' role and add principals we use in headers
+        group_with_UA_admin = self._create_group_with_user_access_administrator_role(self.tenant)
+        group_with_UA_admin.principals.add(self.user_based_principal, self.service_account_principal)
+
+        # Create a group with 'User Access administrator' role and a principal
+        test_group = Group(name="test group", tenant=self.tenant)
+        test_group.save()
+
```
Comment (petracihalova) [inline tests/management/group/test_view.py:3940]:
```suggestion
        test_group = Group.objects.create(name="test group", tenant=self.tenant)

```

**Comment by vbelchio** [comment]
LGTM
Local test were done and passing

**Pull request merged**